### ISSUE TITLE
[MediaBundle] CollectionRepository incompatible with doctrine/orm 2.10

### ISSUE THREAD

**Issue by YetiCGN** [sulu/sulu#6278]
| Q | A
| --- | ---
| Sulu Version | 2.3.5
| PHP Version | 8.0.11
| DB Version | MariaDB 10.4.13 (but irrelevant)

# Actual Behavior

Starting with version 2.10.0, `doctrine/orm` sets a return type for `getMaxResults()` as nullable integer. The CollectionRepository uses `setMaxResults()` with a string, which is accepted but leads to an error when `getMaxResults()` is called.

Stack trace:
```
in vendor/doctrine/orm/lib/Doctrine/ORM/QueryBuilder.php:657 
#0 vendor/sulu/sulu/src/Sulu/Bundle/MediaBundle/Entity/CollectionRepository.php(210): Doctrine\ORM\QueryBuilder->setMaxResults() 
#1 vendor/sulu/sulu/src/Sulu/Bundle/MediaBundle/Collection/Manager/CollectionManager.php(170): Sulu\Bundle\MediaBundle\Entity\CollectionRepository->findCollections() 
#2 vendor/sulu/sulu/src/Sulu/Bundle/MediaBundle/Controller/CollectionController.php(237): Sulu\Bundle\MediaBundle\Collection\Manager\CollectionManager->get() 
#3 vendor/symfony/http-kernel/HttpKernel.php(156): Sulu\Bundle\MediaBundle\Controller\CollectionController->cgetAction() 
#4 vendor/symfony/http-kernel/HttpKernel.php(78): Symfony\Component\HttpKernel\HttpKernel->handleRaw() 
#5 vendor/symfony/http-kernel/Kernel.php(199): Symfony\Component\HttpKernel\HttpKernel->handle() 
#6 public/index.php(66): Symfony\Component\HttpKernel\Kernel->handle() 
#7 {main}
```

# Expected Behavior

Sulu should honor the integer type of the property and cast any string to int before using Doctrine's setters.

# Steps to Reproduce

Reproducible for example with Collections API call: https://localhost:8000/admin/api/collections?fields=title%2CobjectCount%2Cid&flat=true&limit=10&locale=de&page=1&sortBy=title&sortOrder=asc

See also https://github.com/doctrine/orm/issues/9086

# Possible Solutions

Cast `$limit` to int before handing it over to the setter.

**Comment by alexander-schranz** [comment]
This should be fixed in https://github.com/sulu/sulu/pull/6271/files#diff-434625111321abb225365806085a3753cb854d098ed9b4b6dc2001235c7f7bc4. You can try to test it with using `~2.3.5@dev` in your project. Please check also the upgrade.md of the PR if you use doctrine/dbal 3.

**Comment by alexander-schranz** [comment]
Also seems like doctrine/orm is handling this now correctly: https://github.com/doctrine/orm/pull/9079.

**Comment by YetiCGN** [comment]
Yes, but that's also not released yet and if one does a blind `composer update` suddenly the MediaBundle stops working altogether. 🙂

**Comment by YetiCGN** [comment]
> You can try to test it with using `~2.3.5@dev` in your project. 

Confirmed. That fixes the issue even with `doctrine/orm 2.10.0`. We don't use DBAL 3 yet.
